js/exporter/image_creator.js

```diff
@@ -419,31 +419,44 @@ function drawElement(element, context, parentOptions, shared) {
 
     if(!isText) {
         applyFilter(context, options, shared);
-        fillElement(context, options, shared);
+        if(!isImage) {
+            promise = fillElement(context, options, shared);
+        }
         strokeElement(context, options);
     }
 
-    applyGradient(context, options, shared, element);
+    applyGradient(context, options, shared, element, 'linear');
+    applyGradient(context, options, shared, element, 'radial');
 
     context.restore();
 
     return promise;
 }
 
-function applyGradient(context, options, { gradients }, element) {
-    if(gradients.length === 0) {
+function applyGradient(context, options, { linearGradients, radialGradients }, element, type) {
+    const gradients = type === 'linear' ? linearGradients : radialGradients;
+    if(Object.keys(gradients).length === 0) {
         return;
     }
     const id = parseUrl(options.fill);
     if(id && gradients[id]) {
         const box = element.getBBox();
-        const gradient = context.createLinearGradient(box.x, 0, box.x + box.width, 0);
+        const horizontalCenter = box.x + box.width / 2;
+        const verticalCenter = box.y + box.height / 2;
+        const gradient = type === 'linear' ?
+            context.createLinearGradient(box.x, 0, box.x + box.width, 0) :
+            context.createRadialGradient(horizontalCenter, verticalCenter, 0, horizontalCenter, verticalCenter, Math.max(box.height / 2, box.width / 2));
```

Comment:
`Math.max(box.height / 2, box.width / 2)`, can you put this line in a separate variable ? 

At the moment I don't understand why we need the maximum radius

Reply:
>At the moment I don't understand why we need the maximum radius
If we choose minimum radius, then gradient on the narrow element will be small. In the svg label gradient fill all the space, thus I choose maximum radius

Reply:
That's why I suggested putting it in a separate variable, so that there would be no such question.